include/kafka/AdminClient.h

```diff
@@ -292,5 +323,51 @@ AdminClient::listTopics(std::chrono::milliseconds timeout)
     return Admin::ListTopicsResult(names);
 }
 
+inline Admin::DeleteRecordsResult
+AdminClient::deleteRecords(const TopicPartitionOffsets& tpos,
+                           std::chrono::milliseconds timeout) {
+    auto rk_queue = rd_kafka_queue_unique_ptr(rd_kafka_queue_new(getClientHandle()));
+
+    rd_kafka_DeleteRecords_unique_ptr rkDeleteRecords(rd_kafka_DeleteRecords_new(createRkTopicPartitionList(tpos)));
+    rd_kafka_DeleteRecords_t* rk_offsets = rkDeleteRecords.get();
+
+    rd_kafka_DeleteRecords(getClientHandle(), &rk_offsets, 1, nullptr, rk_queue.get());
```

Comment:
How about removing line 332 (i.e. `rd_kafka_DeleteRecords_t* rk_offsets = rkDeleteRecords.get();`), 
thus here could be 
```
rd_kafka_DeleteRecords(getClientHandle(), &rkDeleteRecords.get(), 1, nullptr, rk_queue.get());
```
(thus we could get rid of naming trouble, -- since `rk_offsets` is not quite a good name) 😅

Reply:
I gave it a try, but I find compiler doesn't like this change and reported :(
```
cannot take the address of an rvalue of type 'std::__1::unique_ptr<rd_kafka_DeleteRecords_s, kafka::RkDeleteRecordsDeleter>::pointer' (aka 'rd_kafka_DeleteRecords_s *')
    rd_kafka_DeleteRecords(getClientHandle(), &rkDeleteRecords.get(), 1, nullptr, rk_queue.get());
```

So I renamed this variable to be `rk_del_records` (which is similar with librdkafka's unit test code [https://github.com/edenhill/librdkafka/blob/4837e34bc5d173b10934e874adadf39f98dd2b94/tests/0080-admin_ut.c#L563](here)).  Let me know if you prefer a different name or there is better alternative approach for doing this.

Reply:
using `rk_del_records` sounds good!